### PR TITLE
Add php8.0 to composer.json

### PR DESCRIPTION
**Pull request by Glancu**
Added php 8.0 to composer.json for SonataAdminBundle - continuous-integration/travis-ci/pr

### CODE REVIEW

**Comment by dbu** [comment]
you should add php 8.0 to the build matrix in https://github.com/KnpLabs/KnpMenuBundle/blob/master/.travis.yml as well, so that it is visible that it indeed works.

**Comment by garak** [comment]
Any hint on how we can avoid failure due to `Call to a member function setDeprecated() on null` in Symfony container?
I've got same problem in other bundles too 😔

**Comment by Glancu** [comment]
Travis does not support php 8 :(

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ php:
     - 7.2
     - 7.3
     - 7.4
+    - 8.0
```
Comment (garak) [inline .travis.yml:19]: This currently doesn't work. Use `nightly` instead

**Comment by garak** [comment]
You need to open a similar PR on https://github.com/KnpLabs/KnpMenu

**Comment by Glancu** [comment]
> You need to open a similar PR on https://github.com/KnpLabs/KnpMenu


Now is correctly or i need change php 8.0 in matrix to nightly?

**Comment by garak** [comment]
> > You need to open a similar PR on https://github.com/KnpLabs/KnpMenu
> 
> Now is correctly or i need change php 8.0 in matrix to nightly?

You need to. It's not possible (yet) to use "8.0" in travis

**Comment by Glancu** [comment]
> > > You need to open a similar PR on https://github.com/KnpLabs/KnpMenu
> > 
> > 
> > Now is correctly or i need change php 8.0 in matrix to nightly?
> 
> You need to. It's not possible (yet) to use "8.0" in travis

Okay, today is my first time, when i try to fix errors in public projects so sorry for that

**Comment by garak** [comment]
> Okay, today is my first time, when i try to fix errors in public projects so sorry for that

No need to apologize! Glad to help 😉

**Comment by derrabus** [comment]
> Any hint on how we can avoid failure due to `Call to a member function setDeprecated() on null` in Symfony container?

Should be fixed with #431.

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     ],
     "require": {
         "php": "^7.2 || ^8.0",
-        "knplabs/knp-menu": "^3.0",
+        "knplabs/knp-menu": "^3.2",
```
Comment (garak) [inline composer.json:23]: We should use `^3.2@dev` instead, otherwise we can't get a php8-compatible version

Comment (Glancu) [reply composer.json]: I fix this now, but i have mistake in Scrutinizer and i don't know how i can fix this.

Comment (Glancu) [reply composer.json]: Okay, tests works, but I have error with phpcs-run and i don't know how to fix it. Do you know how to fix it?

Comment (garak) [reply composer.json]: it looks like phpcs still doesn't support php 8

Comment (Glancu) [reply composer.json]: Okay, so we need to wait for support or we can merge this PR?

Comment (garak) [reply composer.json]: There's no hurry

**Comment by Punk-UnDeaD** [comment]
KnpMenu v3.1.2 support PHP 8 already.
Today is a good day for release.

**Pull request merged**